mpi_common.sh

```diff
@@ -0,0 +1,45 @@
+# MPI helper shell functions
+function check_efa_ompi {
+    out=$1
+    grep -q "mtl:ofi:prov: efa" $out
+    if [ $? -ne 0 ]; then
+        echo "efa provider not used with Open MPI"
+        exit 1
+    fi
+}
+
+function check_efa_impi {
+    out=$1
+    grep -q "libfabric provider: efa" $out
+    if [ $? -ne 0 ]; then
+        echo "efa provider not used with Intel MPI"
+        exit 1
+    fi
+}
+
+function ompi_setup {
+    . /etc/profile.d/efa.sh
+    export OMPI_MCA_mtl_base_verbose=100
```

Comment:
`check_efa_ompi` would fail if EFA isn't picked up, but can we also force the use of the MTL OFI with an MCA param so the MPI job itself fails?

Reply:
I guess my concern is if it picks up another libfabric provider and the job continues so I think we need both the check here and the MCA param.

Reply:
Agreed, we need both, hence the _also_.

Reply:
I see, sorry. I'll open another PR with that change soon.